### PR TITLE
feat: Use StartingInventorySystem via prefab delta

### PR DESCRIPTION
**Pull request by askneller**
Switch ```MetalRenegades``` to using ```StartingInventorySystem```.

**How To Test**

Load a game with ```MetalRenegades``` as the gameplay module and verify that the starting inventory is the same (may be in a different order).

### CODE REVIEW

**Comment by askneller** [comment]
@skaldarnar  regarding your comment on the CombatSystem PR, I think it might be a good idea to move the gameplay module related parts of ```CombatSystem``` (specifically ```CombatStartingInventory```, ```NPCSpawnSystem```, and the world generation related code) to a separate gameplay module (e.g. ```CombatSystemGameplay```). That way we separate the gameplay/demonstration features from the actual combat related code. Thoughts?

File: src/main/java/org/terasology/metalrenegades/PlayerInventorySystem.java [hunk 1]
```diff
@@ -72,19 +67,9 @@
 
     @Override
     public void postBegin() {
+        // Ensure that CombatSystem does not spawn its own starting inventory.
```
Comment (skaldarnar) [inline src/main/java/org/terasology/metalrenegades/PlayerInventorySystem.java:70]: Can this whole method be removed? What does `CombatStartingInventory` do now that it does not need to do anything?

**Review by skaldarnar — COMMENTED** [review]
I agree with moving the starting inventory out of the non-gameplay module `CombatSystem`. I'm not sure whether we should introduce a new game mode just for showcasing the items, though.

Maybe this *demo* features could be handled by command in the modules (with cheat permission level)? Something like `give combatSystemCollection` or even `give collection combatSystem`?

File: src/main/java/org/terasology/metalrenegades/PlayerInventorySystem.java (hunk 1 above)
Comment (askneller) [reply src/main/java/org/terasology/metalrenegades/PlayerInventorySystem.java:70]: As it currently stands, ```CombatStartingInventory``` runs for any module that has ```CombatSystem``` as a dependency. If a module that depends on ```CombatSystem``` does not change the starting inventory then the inventory specified in ```CombatSystem``` will be added regardless.  

As a test you can run this branch and comment out the entire ```MetalRenegades:PlayerInventorySystem.postBegin()``` method, and you should see that the ```CombatSystem``` inventory is added <em>in addition</em> to the ```MetalRenegades``` starting inventory.

I hope I'm making sense.

Comment (skaldarnar) [reply src/main/java/org/terasology/metalrenegades/PlayerInventorySystem.java:70]: Absolutely! I now did understand that you're actually resetting the starting inventory which would have contained combat items to not have any items initally. 
The previous code did overide the default starting inventory of CombatSystem with what MetalRenegades has.

The whole code block can be removed once we removed the default starting inventory from CombatSystem module (and move it to a command or some other gameplay module), is that correct?

**Review by skaldarnar — APPROVED** [review]

File: src/main/java/org/terasology/metalrenegades/PlayerInventorySystem.java (hunk 1 above)
Comment (askneller) [reply src/main/java/org/terasology/metalrenegades/PlayerInventorySystem.java:70]: Yes, I believe that's correct.

**Pull request merged**